### ISSUE TITLE
drloc loss formulation

### ISSUE THREAD

**Issue by julien-gadonneix** [yhlleo/VTs-Drloc#11]
The loss is defined like this:

```
def cal_selfsupervised_loss(outs, args, lambda_drloc=0.0):
    loss, all_losses = 0.0, Munch()
    if args.TRAIN.USE_DRLOC:
        if args.TRAIN.DRLOC_MODE == "l1": # l1 regression constraint
            reld_criterion = relative_constraint_l1 
        elif args.TRAIN.DRLOC_MODE == "ce": # cross entropy constraint
            reld_criterion = relative_constraint_ce
        elif args.TRAIN.DRLOC_MODE == "cbr": # cycle-back regression constaint: https://arxiv.org/pdf/1904.07846.pdf
            reld_criterion = relative_constraint_cbr
        else:
            raise NotImplementedError("We only support l1, ce and cbr now.")

        loss_drloc = 0.0
        for deltaxy, drloc, plane_size in zip(outs.deltaxy, outs.drloc, outs.plz):
            loss_drloc += reld_criterion(deltaxy, drloc) * lambda_drloc
        all_losses.drloc = loss_drloc.item()
        loss += loss_drloc

    return loss, all_losses
```

However, in the paper, it is formulated like this:
![Image](https://github.com/user-attachments/assets/edd830bc-4342-4cd2-92a5-b3a15642d038)

In order to approximate the expected value, shouldn't we divide by the number of sampled pairs? Otherwise, this auxiliary loss grows linearly with the number of sampled pairs.

**Comment by yhlleo** [comment]
@julien-gadonneix 

Yes, it would be better to divide by the number of sampled pairs. In fact, it is inside in the `reld_criterion` function. For example, if we set `args.TRAIN.DRLOC_MODE == "l1"`, we calculate the loss by `F.l1_loss(deltaxy, predxy)`.  As you can see, the default reduction manner is "mean":

<img width="855" alt="Image" src="https://github.com/user-attachments/assets/0484a07e-371e-40d8-b8d2-f27d2f62c3af" />

**Comment by julien-gadonneix** [comment]
Yes you're right !
Sorry I forgot that the output was like this:
```
outs.drloc = [drloc_feats]
outs.deltaxy = [deltaxy]
```
